Application/Frontend/pages/api/Get_userGroup.ts

```diff
@@ -0,0 +1,50 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { MongoClient } from 'mongodb';
+import { getMongoDbUri } from '@/lib/dbConfig';
+
+/**
+ * API endpoint handler for fetching message history from the MongoDB database.
+ * This handler responds to POST requests with a specific channel key and retrieves
+ * the corresponding chat history.
+ *
+ * @param {NextApiRequest} req The incoming HTTP request. Expects a POST method
+ * with a body containing the 'channelKey' to identify the chat whose history is being requested.
+ * @param {NextApiResponse} res The outgoing HTTP response. Returns the chat history on success,
+ * an error message on failure, and a method not allowed message if the HTTP method is not POST.
+ */
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  // Only respond to POST requests; other methods are not supported for this endpoint.
+  if (req.method === 'POST') {
+    const { channelKey } = req.body; // Extract the channelKey from the request body.
+
+    let client: MongoClient | null = null; // Define a variable to hold the MongoDB client.
+
+    try {
+      client = new MongoClient(getMongoDbUri()); // Initialize the MongoDB client.
+      await client.connect(); // Establish a connection to the database.
+      const db = client.db('Accord'); // Access the 'Accord' database.
+      const chatsCollection = db.collection('Chats'); // Access the 'Chats' collection.
+
+      const chat = await chatsCollection.findOne({ channelKey: channelKey }); // Find the chat with the given channel key.
+
+      // If the chat is found, return its message history; otherwise, return a 404 error.
+      if (chat) {
+        res.status(200).json({ messageHistory: chat.messageHistory });
```

Comment:
Why are you returning the message history? Not only is the name of the API endpoint so poorly constructed (`Get_userGroup`?), but this endpoint in itself is a duplicate of `get-message-history` that has already existed for weeks in the code base.